### PR TITLE
Feature cosmos coins

### PR DESCRIPTION
**Pull request by BitHighlander**
Terra Kava Secret support

### CODE REVIEW

File: integration/src/kava/tx01.mainnet.kava.description.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (kaladinlight) [inline integration/src/kava/tx01.mainnet.kava.description.json:1]: ?

File: packages/hdwallet-core/src/cardano.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import { addressNListToBIP32, slip44ByCoin } from "./utils";
```
Comment (kaladinlight) [inline packages/hdwallet-core/src/cardano.ts:1]: Not sure if cardano supposed to be in this pr

File: integration/src/kava/tx01.testnet.kava.txid.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (kaladinlight) [inline integration/src/kava/tx01.testnet.kava.txid.json:1]: Is this being used?

File: packages/hdwallet-native/package.json [hunk 4]
```diff
@@ -14,7 +14,10 @@
     "prepublishOnly": "yarn clean && yarn build"
   },
   "dependencies": {
+    "@cosmostation/cosmosjs": "^0.9.5",
+    "@pioneer-platform/secret-tx-builder": "6.2.3",
```
Comment (kaladinlight) [inline packages/hdwallet-native/package.json:18]: curious what we need this for that isn't availble in `secretjs-offline`

File: packages/hdwallet-ledger/src/ledger.ts [hunk 5]
```diff
@@ -137,6 +137,14 @@ export class LedgerHDWalletInfo implements core.HDWalletInfo, core.BTCWalletInfo
...
   _supportsFioInfo: boolean = false;
   _supportsThorchainInfo: boolean = false;
+  _supportsSecret: boolean = false;
+  _supportsSecretInfo: boolean = false;
+  _supportsKava: boolean = false;
+  _supportsKavaInfo: boolean = false;
+  _supportsTerra: boolean = false;
+  _supportsTerraInfo: boolean = false;
+  _supportsCardano: boolean = false;
+  _supportsCardanoInfo: boolean = false;
```
Comment (kaladinlight) [inline packages/hdwallet-ledger/src/ledger.ts:147]:
```suggestion
  _supportsSecretInfo: boolean = false;
  _supportsKavaInfo: boolean = false;
  _supportsTerraInfo: boolean = false;
  _supportsCardanoInfo: boolean = false;
```

File: packages/hdwallet-keepkey/src/keepkey.ts [hunk 6]
```diff
@@ -680,7 +683,16 @@ export class KeepKeyHDWallet implements HDWallet, BTCWallet, ETHWallet, DebugLin
   _supportsBinance: boolean = true;
   _supportsEos: boolean = true;
   _supportsFio: boolean = false;
+  _supportsThorchainInfo: boolean = true;
```
Comment (kaladinlight) [inline packages/hdwallet-keepkey/src/keepkey.ts:686]: Weird that `_supportsInfo` has been added to all of the wallets...

File: packages/hdwallet-core/src/utils.ts [hunk 7]
```diff
@@ -139,6 +140,10 @@ export function slip44ByCoin(coin: Coin): number {
     Eos: 194,
     Fio: 235,
     Thorchain: 931,
+    Cardano: 1815,
+    Secret: 529,
+    Terra: 118, //match atom
```
Comment (kaladinlight) [inline packages/hdwallet-core/src/utils.ts:145]: Why shouldn't it be set to it's own slip44?

File: integration/src/terra/tx01.mainnet.terra.txid.json [hunk 8]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (kaladinlight) [inline integration/src/terra/tx01.mainnet.terra.txid.json:1]: Delete?

File: integration/src/secret/tx01.mainnet.secret.txid.json [hunk 9]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (kaladinlight) [inline integration/src/secret/tx01.mainnet.secret.txid.json:1]: Used?

File: integration/src/kava/kava.ts [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+      });
+    }, TIMEOUT);
+
+    test(
+      "kavaGetAccountPaths()",
+      () => {
+        if (!wallet) return;
+        const paths = wallet.kavaGetAccountPaths({ accountIdx: 0 });
+        expect(paths.length > 0).toBe(true);
+        expect(paths[0].addressNList[0] > 0x80000000).toBe(true);
```
Comment (kaladinlight) [inline integration/src/kava/kava.ts:39]: Test for more legit values? actual purpose, coin type and account index

File: integration/src/kava/kava.ts [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+      },
+      TIMEOUT
+    );
+
+    test(
+      "describePath() kava",
+      async () => {
+        if (!wallet) return;
+        expect(
```
Comment (kaladinlight) [inline integration/src/kava/kava.ts:48]: these tests aren't testing anything

File: integration/src/secret/secret.ts [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+
+const MNEMONIC12_NOPIN_NOPASSPHRASE = "alcohol woman abuse must during monitor noble actual mixed trade anger aisle";
+
+const TIMEOUT = 60 * 1000;
+
+/**
+ *  Main integration suite for testing SecretWallet implementations' Secret support.
+ */
+export function secretTests(get: () => { wallet: core.HDWallet; info: core.HDWalletInfo }): void {
```
Comment (kaladinlight) [inline integration/src/secret/secret.ts:15]: similar comments from kava here

File: integration/src/terra/terra.ts [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+
+
+const MNEMONIC12_NOPIN_NOPASSPHRASE = "alcohol woman abuse must during monitor noble actual mixed trade anger aisle";
+
+const TIMEOUT = 60 * 1000;
+
+/**
+ *  Main integration suite for testing TerraWallet implementations' Terra support.
+ */
+export function terraTests(get: () => { wallet: core.HDWallet; info: core.HDWalletInfo }): void {
```
Comment (kaladinlight) [inline integration/src/terra/terra.ts:14]: and here

File: packages/hdwallet-portis/src/portis.ts [hunk 14]
```diff
@@ -76,6 +76,14 @@ export class PortisHDWallet implements HDWallet, ETHWallet, BTCWallet {
   _supportsFio: boolean = false;
   _supportsThorchainInfo: boolean = false;
   _supportsThorchain: boolean = false;
+  _supportsSecretInfo: boolean = false;
+  _supportsSecret: boolean = false;
+  _supportsKava: boolean = false;
+  _supportsKavaInfo: boolean = false;
+  _supportsTerra: boolean = false;
+  _supportsTerraInfo: boolean = false;
+  _supportsCardano: boolean = false;
```
Comment (kaladinlight) [inline packages/hdwallet-portis/src/portis.ts:85]: Cardano

File: packages/hdwallet-native/src/kava.ts [hunk 15]
```diff
@@ -0,0 +1,123 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import { BIP32Interface } from "bitcoinjs-lib";
+import * as bitcoin from "bitcoinjs-lib";
```
Comment (kaladinlight) [inline packages/hdwallet-native/src/kava.ts:3]: single import per package
```suggestion
import { BIP32Interface } from "bitcoinjs-lib";
import * as bitcoin from "bitcoinjs-lib";
```

File: packages/hdwallet-native/src/kava.ts [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+import { NativeHDWalletBase } from "./native";
+import { getNetwork } from "./networks";
+import { toWords, encode } from "bech32";
+import CryptoJS, { RIPEMD160, SHA256 } from "crypto-js";
+import util from "./util";
+const cosmosjs = require("@cosmostation/cosmosjs");
```
Comment (kaladinlight) [inline packages/hdwallet-native/src/kava.ts:9]: import > require

**Review by kaladinlight — APPROVED** [review]

File: packages/hdwallet-native/src/cosmos.ts [hunk 17]
```diff
@@ -45,7 +44,7 @@ export function MixinNativeCosmosWallet<TBase extends core.Constructor<NativeHDW
   return class MixinNativeCosmosWallet extends Base {
     _supportsCosmos = true;
 
-    #wallet: BIP32Interface;
+    #wallet: bitcoin.BIP32Interface;
```
Comment (mvpratt) [inline packages/hdwallet-native/src/cosmos.ts:47]: what this?

File: packages/hdwallet-native/src/native.ts [hunk 18]
```diff
@@ -118,6 +127,17 @@ class NativeHDWalletInfo
       case "trune":
       case "thorchain":
         return core.thorchainDescribePath(msg.path);
+      case "secret":
+      case "scrt":
+      case "tscrt":
```
Comment (mvpratt) [inline packages/hdwallet-native/src/native.ts:132]: why multiple names for the same chain?

**Pull request merged**